wpiutil/src/main/java/edu/wpi/first/util/telemetry/TelemetryTable.java

```diff
@@ -0,0 +1,345 @@
+
+package edu.wpi.first.util.telemetry;
+
+import edu.wpi.first.util.protobuf.Protobuf;
+import edu.wpi.first.util.protobuf.ProtobufSerializable;
+import edu.wpi.first.util.struct.Struct;
+import edu.wpi.first.util.struct.StructSerializable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import us.hebi.quickbuf.ProtoMessage;
+
+/**
+ * Telemetry sends information from the robot program to dashboards, debug tools, or log files.
+ *
+ * <p>For more advanced use cases, use the NetworkTables or DataLog APIs.
+ */
+public final class TelemetryTable {
+  private final String m_path;
+  private final ConcurrentMap<String, TelemetryTable> m_tablesMap = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, TelemetryEntry> m_entriesMap = new ConcurrentHashMap<>();
+
+  /**
+   * Constructs a telemetry table.
+   *
+   * @param path path with trailing "/".
+   */
+  public TelemetryTable(String path) {
+    m_path = path;
+  }
+
+  /**
+   * Gets a child telemetry table.
+   *
+   * @param name table name
+   * @return table
+   */
+  public TelemetryTable getTable(String name) {
+    return m_tablesMap.computeIfAbsent(name, k -> new TelemetryTable(m_path + k + "/"));
+  }
+
+  /**
+   * Gets a telemetry entry.
+   *
+   * @param name name
+   * @return entry
+   */
+  private TelemetryEntry getEntry(String name) {
+    return m_entriesMap.computeIfAbsent(name, k -> TelemetryRegistry.getEntry(m_path + k));
+  }
+
+  /**
+   * Indicates duplicate values should be preserved. Normally duplicate values are ignored.
+   *
+   * @param name the name
+   */
+  public void keepDuplicates(String name) {
+    getEntry(name).keepDuplicates();
+  }
+
+  /**
+   * Sets property for a value. Properties are stored as a key/value map.
+   *
+   * @param name the name
+   * @param key property key
+   * @param value property value
+   */
+  public void setProperty(String name, String key, String value) {
+    getEntry(name).setProperty(key, value);
+  }
+
+  /**
+   * Sets custom data type for a value. Generally not necessary.
+   *
+   * @param name the name
+   * @param typeString type string
+   */
+  public void setTypeString(String name, String typeString) {
+    getEntry(name).setTypeString(typeString);
+  }
+
+  /**
+   * Logs a generic object.
+   *
+   * @param name the name
+   * @param value the value
+   */
+  public void log(String name, Object value) {
+    if (value instanceof TelemetryLoggable) {
+      ((TelemetryLoggable) value).log(getTable(name));
+    } else if (value instanceof StructSerializable) {
+      // TODO: introspection
+      // getEntry(name).logStruct((StructSerializable) value);
+    } else if (value instanceof ProtobufSerializable) {
+      // TODO: introspection
+      // getEntry(name).logProtobuf((ProtobufSerializable) value);
+    } else if (value instanceof Boolean) {
+      log(name, ((Boolean) value).booleanValue());
```

Comment:
```suggestion
    } else if (value instanceof Boolean b) {
      log(name, b.booleanValue());
```
I know this is still a draft, but I figured I might as well mention that this is an option.
(See 14.30 and 15.20.2 in https://docs.oracle.com/javase/specs/jls/se17/jls17.pdf if you want to see details)